service/history/replication/fx.go

```diff
@@ -119,18 +120,32 @@ func ReplicationTaskExecutorProvider() TaskExecutorProvider {
 func ReplicationStreamSchedulerProvider(
 	config *configs.Config,
 	logger log.Logger,
+	queueFactory ctasks.SequentialTaskQueueFactory[TrackableExecutableTask],
 ) ctasks.Scheduler[TrackableExecutableTask] {
 	return ctasks.NewSequentialScheduler[TrackableExecutableTask](
 		&ctasks.SequentialSchedulerOptions{
 			QueueSize:   config.ReplicationProcessorSchedulerQueueSize(),
 			WorkerCount: config.ReplicationProcessorSchedulerWorkerCount,
 		},
 		WorkflowKeyHashFn,
-		NewSequentialTaskQueue,
+		queueFactory,
 		logger,
 	)
 }
 
+func SequentialTaskQueueFactoryProvider(
+	logger log.Logger,
+	metricsHandler metrics.Handler,
+	config *configs.Config,
+) ctasks.SequentialTaskQueueFactory[TrackableExecutableTask] {
+	return func(task TrackableExecutableTask) ctasks.SequentialTaskQueue[TrackableExecutableTask] {
+		if config.EnableReplicationTaskBatching() {
```

Comment:
Feature flag here maybe not enough? If the task for same workflow keep coming, then no new queue will be created, and we need to restart history pod to pick up the change.